Improved many UI elements, better iPad split view sizing, new URL schemes, and a variety of bug fixes

ShihabM opened the pull request:
- Added account usernames under profile display pictures in the top-left quick switcher
- Added the ability to go to the settings section via the top-left quick switcher
- Added contextual headers to the top-left quick switcher to better inform users about what the different parts of the switcher are for
- An indication of who boosted the toot is now displayed above the respective toot rather than below it, and uses an improved boost icon along with display names rather than usernames for clarity by default
- Added the option to select whether boosted text displays display names or usernames (can be changed via settings)
- Boosted toots now also display a small display picture of the user who boosted the toot in question
- Added a network connectivity alert that's displayed when the app can't connect to the internet via WiFi or Cellular
- Added a hardware keyboard command for the iPad to go to the settings section
- Added a hardware keyboard command for the iPad to close the toot detail view
- Added a new URL scheme to send a toot with text filled in already (bypassing the visual composition screen) - this can be found in the settings section's URL schemes overview section (toot=hello to send a toot with the text 'hello')
- Added a new URL scheme to go to the app's settings section
- Added drag deletion of polls from the toot compose screen (simply drag the poll icon to the red area in the same way as dragging image icons to the red area)
- Renamed the 'Federated' timeline segment to 'All' to better suggest to the user what the timeline contains (especially for those not familiar with the federated universe)
- Adjusted the iPad column ratio to give more space to the primary column and make better use of available space (50/50 by default now)
- Added a button to the top-left of the iPad detail view to adjust the split ratio of the right and left panes with more granularity (20%, 25%, 30%, 35%, 40%, 45%, 50%)
- Fixed issue where dragging to remove images from the toot composition screen would also remove the bar above the keyboard
- Fixed issue where removing a poll from a composed toot wouldn't always remove it
- Fixed issue where the followers/following section wouldn't load in or display any profiles
- Fixed issue where tapping certain toots in timelines would crash the app
- Fixed issue where videos and GIFs wouldn't be displayed in toot detail views
- Fixed issue where toot details wouldn't display multiple images if the toot contained imore than one image on the iPad app
- Fixed issue where tapping profile icons in the iPad's detail split view wouldn't display the tapped profile
- Fixed issue where tapping some toots on the iPad would sometimes crash the app
- Fixed issue where pulling to refresh the right-hand side pane on the iPad when it wasn't populated would crash the app
- Fixed layout issue where segments on the iPad would be shifted to one side or squashed into other elements on the screen
- Fixed issue where tapping boosts/likes counts in the iPad's detail split view wouldn't display the toot's boosts or likes
- Fixed issue where tapping mentions or hashtags in the iPad's detail split view wouldn't display the respective user or hashtag
- Minor UI adjustments
- Minor bug fixes

Commit:
iPad redesign, WIP

Commit:
iPad redesign, WIP

Commit:
iPad redesign, WIP

Commit:
iPad redesign, WIP

Commit:
Initial polls logic, WIP

Commit:
Initial polls logic, WIP

Commit:
iPad redesign, WIP

Commit:
Initial polls logic, WIP

Commit:
Initial polls logic, WIP

Commit:
Initial polls logic, WIP

Commit:
Initial polls logic, WIP

Commit:
Quick switcher changes, text changes, etc, WIP

Commit:
Minor changes, WIP

Commit:
Minor changes, WIP

Commit:
Minor changes, WIP

Commit:
Minor changes, WIP

Commit:
Minor changes, WIP

Commit:
Minor fixes, WIP

barrettbreshears reviewed (approved):
looks great going to merge this in 👍

Commit:
Minor changes, WIP

Commit:
Minor changes, WIP

ShihabM commented:
Wanted to add the following further changes to this PR - they've been added in now:

- Added the ability to add a filter to images that were added to a toot via the compose screen, by tapping the added image and selecting the filter option
- Swipe right and left on the main filtered image to change the filter, or scroll through the filters below the image to do the same
- Added option to clear text styling from the typed text when composing toots, accessible via the compose screen's ••• more button's text styles option or by long-holding any selected text to get the contextual pop-up and swiping all the way to the right
- Recreated link previews in toot details to give improved details for the link, with more contextual links that don't also include random URLs
- Tapping a video link now plays the video natively rather than opening it in the in-app Safari view
- Added the ability to play YouTube videos natively in the app by tapping the YouTube link preview
- Selected iPad screen split ratios are now persisted and retained across sessions
- Added alert when a GIF or video couldn't be added when composing toots to alert the user that it failed to add it, this also prevents accidentally posting these as images
- Added the current instance's image to the instance details section accessible via the ••• more button on the user profile section
- Tapping profiles, mentions, hashtags, and other elements on the iPad's detail section opens the respective views in the primary column now
- The scrollable media picker in the toot composition view now also displays videos with a small icon to indicate that it's a video
- Tapping a video in the toot composition screen is now more contextually aware and displays relevant options rather than those related to images
- Added reply counts in toot details above the replies
- Added a confirmation screen when setting direct visibility when composing toots to alert the user that all mentioned users will still see the message
- Amended the toot composition URL scheme to present the toot composer with the filled in text rather than automatically posting on your behalf for security reasons (bypassing the visual composition screen) - this can be found in the settings section's URL schemes overview section (toot=hello to send a toot with the text 'hello')
- Removed the smallest split ratios for the iPad as they make viewing content almost impossible and squash elements on top of each other
- Fixed issue where the following/followers section would display only followers for both segments and sometimes display duplicates
- Fixed layout issue where censored overlays wouldn't properly cover boosted toot usernames
- Fixed issue where tapping profiles, mentions, hashtags, and other elements on the iPad's detail section would crash the app
- Fixed issue where attempting to share content (text and links) on the iPad would crash the app
- Fixed issue where link previews would display content that wasn't useful or usable
- Fixed resizing issue with user profiles on the iPad that would occur when loading them in
- Fixed issue where certain sections on the iPad would display a title that would overlap segments and other elements
- Fixed issue where the option to resume composing a toot would pop up every time after composing or discarding a toot and reentering the app regardless of whether it failed to post the toot or not
- Fixed issue where certain characters wouldn't be styled when using text styling whilst composing a toot
- Fixed issue where the same account would sometimes show up multiple times when being added to the app
